### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by cmg2213**

### CODE REVIEW

File: assignment5.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Caleb Gershengorn
+
+1:
+You would call the function by typing fxn(a,b,c) where a,b,c would be constants or variables inserted by the user.
+It might look something like this:
+void main(){
```
Comment (oldclesleycode) [inline assignment5.txt:6]: right, don't forget the main declaration before the main!

File: assignment5.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	printf("%d\n", fxn(a,b,c); \\calls the fxn function and prints the result using %d because fxn calls for an int
+	\\This would call the fxn and use the three values the user entered to run the function
+	}
+
+2:
+Iteration is running a loop a certain number of times in order to achieve a certain result, updating the value of a variable every time the loop runs and returning a new value when the loop runs.
+Recursion is a process that involves a function calling itself a certain number of times, updating a counting variabl but not updating the variable that you want to change until the end.
+Once the recursive function reaches its end, it remembers how many times it has run, and runs itself that many times, updating the variable each time.
+Loops are preferable in instances when the variables are simple to sort through, but recursion is useful when you might have to check a certain value again or when something is a function of itself, like a tree.
+Recursion is often not preferable because calling a function takes more space in memory than running an iteration, making it less efficient usually.
```
Comment (oldclesleycode) [inline assignment5.txt:22]: right, complexity is key! and now that we're learning about memory management and data structures, you can further understand this by understanding that each time a function is called, a stack is initialized. So having a recursive function can be very memory ineffective.

File: assignment5.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+3:
+A compiler translates the code that you write into machine language that the computer can understand.
+It first reads the preprosessor directives, which tell it to import certain libraries.
+It reads top to bottom, so it starts with either the functions that are defined at the top or the prototype functions put as placeholders to be defined later on.
+The compiler reads the data-type of the function and checks to make sure at the end of the function that that same data-type is returned, or an error pops up.
+Additionally, it stores anything that was told to be stored along the way, such as global variables and constants, and adjusts the data types to their correct storage class
+Finally, it turns all of this into machine language, so that it can be run, and prints out any errors that came along the way as error messages, explaining why it couldn't compile.
```
Comment (oldclesleycode) [inline assignment5.txt:30]: Perfect!

File: assignment5.txt (hunk 1 above)
Comment (oldclesleycode) [inline assignment5.txt:6]: oop, main's type should actually be int.

File: loopFibonacci.c [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	int first;
+	int second;
+	int next;
+	first = 0;
+	second = 1;
+               first=second;
+                second=next;
+		return next;
+}
+void main(){
```
Comment (lesley2958) [inline loopFibonacci.c]: should be int main()!

File: loopFibonacci.c [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	int b;
+	int i;
+        printf("How many terms: \n");
+	scanf("%d\n",&b);
+	for (i=0;i<b;i++)
+        {
+	printf("%d\n", loopFibonacci(b));
+	}
```
Comment (lesley2958) [inline loopFibonacci.c]: there's no return 0; (which is needed as part of convention)

File: loopFibonacci.c [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
```
Comment (lesley2958) [inline loopFibonacci.c]: You should comment your code! Your code entered a loop that wouldn't really exit and then just output 0s.

Comment (oldclesleycode) [inline loopFibonacci.c:1]: You should comment your code! Your code entered a loop that wouldn't really exit and then just output 0s.

File: loopFibonacci.c (hunk 4 above)
Comment (oldclesleycode) [inline loopFibonacci.c:14]: should be int main() - standard convention.

File: loopFibonacci.c [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+		return next;
+}
+void main(){
+	int b;
+	int i;
+        printf("How many terms: \n");
+	scanf("%d\n",&b);
+	for (i=0;i<b;i++)
+        {
+	printf("%d\n", loopFibonacci(b));
```
Comment (oldclesleycode) [inline loopFibonacci.c:21]: No return 0; (again, convention)

File: recurseFibonacci.c [hunk 8]
```diff
@@ -0,0 +1,24 @@
+#include <stdio.h>
+//I cannot get this code to print the fibonacci sequence without having a zero case in the function and then it always prints zero as the first term
+int fibonacci(int N){
+	if (N==1)
+	{
+	return 1;
+	}
+	else
+	{
+	return fibonacci(N-1) + fibonacci(N-2);
```
Comment (oldclesleycode) [inline recurseFibonacci.c:10]: this doesn't really do much - where is the adding of the two previous elements being done?

File: recurseFibonacci.c [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+	}
+}
+void main(){
```
Comment (oldclesleycode) [inline recurseFibonacci.c:13]: Again, C standard's are to make main of type int and have a return 0;

**Pull request closed without merging**